server/recovery/goroutine.go

```diff
@@ -0,0 +1,20 @@
+package recovery
+
+import (
+	"fmt"
+	"runtime/debug"
+)
+
+// Go runs a goroutine with a recovery handler that prevents the plugin from crashing while logging
+// the panic and traceback.
+func Go(name string, logError func(msg string, keyValuePairs ...any), callback func()) {
```

Comment:
Do we not have direct access to `logError` here? Or are there cases where we would want to use a different method than what's provided by the Plugin API?

Reply:
Unfortunately, we don't have global access to the `p.API` that contains`LogError` that routes the error through the RPC protocol with the server. The ergonomics of this pattern here isn't ideal, but I'm hoping to revisit the overall strategy with logging to at least have a generic logging interface passed around vs. this oddly specific callback.

Reply:
Gotcha, so the plan would be to always use `logError`.